lib/pool.js

```diff
@@ -14,28 +14,33 @@ const kQueue = Symbol('queue')
 const kDestroyed = Symbol('destroyed')
 const kClosedPromise = Symbol('closed promise')
 const kClosedResolve = Symbol('closed resolve')
+const kOptions = Symbol('options')
+const kUrl = Symbol('url')
+const kOnDrain = Symbol('onDrain')
+const kOnConnect = Symbol('onConnect')
+const kOnDisconnect = Symbol('onDisconnect')
 
 class Pool extends EventEmitter {
-  constructor (url, {
-    connections,
-    ...options
-  } = {}) {
+  constructor (url, options = {}) {
     super()
 
-    if (connections != null && (!Number.isFinite(connections) || connections <= 0)) {
+    const { connections } = options
+
+    if (connections != null && (!Number.isFinite(connections) || connections < 0)) {
       throw new InvalidArgumentError('invalid connections')
     }
 
+    this[kUrl] = url
+    this[kOptions] = JSON.parse(JSON.stringify(options))
```

Comment:
What scenario is this covering? Failing if `options` is not an object?

Reply:
it's to guard against mutations

Reply:
This is also a known performance issue